packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2471,9 +2539,17 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
   @override
   final bool allowNameWrap;
 
+  bool _isExpandingPropertyValuesForJson(DiagnosticsSerialisationDelegate delegate) {
+    return getProperties().isEmpty && delegate.expandPropertyValues && value is Diagnosticable;
+  }
+
   @override
-  Map<String, Object> toJsonMap() {
-    final Map<String, Object> json = super.toJsonMap();
+  Map<String, Object> toJsonMap(DiagnosticsSerialisationDelegate delegate) {
+    if (_isExpandingPropertyValuesForJson(delegate)) {
```

Comment:
this seems a bit odd. I would expect this is the delegate that should be used when expanding the properties of the value not for everything.
This could do strange things for nodes with zero properties, a Diagnosticable value and multiple children. Granted that is a bit of an edge case but the behavior would be quite surprising.

Reply:
I don't understand what you mean by this? What should the delegate do? I have changed the logic here a little bit.